### DISCUSSION TITLE
How to lower the tab and compartment dividers? They are interfering with stacking.

### DISCUSSION THREAD

**Discussion by GCheung55** [kennetek/gridfinity-rebuilt-openscad#200] [Q&A]
When a tab or compartment divider is present, it acts as a shelf for stacked bins, preventing the bin from properly seating. This poses a problem causing the stacked bins to be higher than expected and reducing bin stability.

![image](https://github.com/kennetek/gridfinity-rebuilt-openscad/assets/46033/6638344c-e641-4a3a-995f-af2cc3c588d8)

How might I go about lowering the tab and compartment divider heights?

**Comment by GCheung55** [comment]
I've found that at https://github.com/kennetek/gridfinity-rebuilt-openscad/blob/main/gridfinity-rebuilt-utility.scad#L131, I can add enough clearance by reducing the `block_bottom` height with `-0.9`.

Note that this change would affect bins without the lip, so some additional logic would need to be added to only apply for the lip.

```
    $block_bottom_height = h0 == 0 ? $dh - 0.1 - (sl == 0 ? 0.9 : 0) : h0;
    difference() {
        color("firebrick")
        block_bottom($block_bottom_height, gx, gy, l);
        children();
    }
```

Perhaps this clearance or offset would be better as a constant?

**Reply by theshining97** [reply]
I appreciate you working on this.  I've had the same issue, of course.  Regarding your code, is `sl == 0` a flag for when `enable_lip`  disabled?  

If so, is `sl` defined elswhere in ~utility.scad?
